### DISCUSSION TITLE
...

### DISCUSSION THREAD

**Discussion by gillerby** [socketio/socket.io#4022] [Q&A] [answered]
...

**Comment by darrachequesne** [comment, marked as answer]
If your website is located at `site.net/chat`, you need to customize the `path` option on the client side:

```html
<script src="https://cdn.socket.io/4.1.3/socket.io.js"></script>
<script>
  var socket = io({
    path: "/chat/socket.io"
  });
</script>
```

Reference: https://socket.io/docs/v4/client-initialization/#path

**Reply by darrachequesne** [reply]
It depends if the path is rewritten or not. If it's left as is, yes you'll need to change the path on the server side, which gives in your case:

```js
const express = require('express');
const app = express();
const http = require('http');
const server = http.createServer(app);
const { Server } = require("socket.io");
const io = new Server(server, {
  path: "/chat/socket.io"
});
```

**Reply by darrachequesne** [reply]
I think your nginx configuration is missing a proxy_pass part.

Please check the documentation here: https://socket.io/docs/v4/reverse-proxy/#NginX
